From df5c5c9d1461740a01818d2da6a89e8900831612 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hugo=20Beauz=C3=A9e-Luyssen?= <hugo@beauzee.fr>
Date: Tue, 19 Jul 2022 11:24:17 +0200
Subject: [PATCH 67/85] opengl: interop_sw: Force use of GL_LUMINANCE for webgl

WebGL doesn't handle our combination of internal format & formats the
same way opengl(es) does, which leads to GL_INVALID_OPERATION being
returned when the interop initializes
---
 modules/video_output/opengl/interop_sw.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/modules/video_output/opengl/interop_sw.c b/modules/video_output/opengl/interop_sw.c
index 6669559b21..d87e20c2aa 100644
--- a/modules/video_output/opengl/interop_sw.c
+++ b/modules/video_output/opengl/interop_sw.c
@@ -587,11 +587,12 @@ opengl_interop_generic_init(struct vlc_gl_interop *interop, bool allow_dr)
         || vlc_gl_HasExtension(&extension_vt, "GL_EXT_unpack_subimage");
 
     /* RG textures are available natively since OpenGL 3.0 and OpenGL ES 3.0 */
-    priv->has_texture_rg = vlc_gl_GetVersionMajor(&extension_vt) >= 3
+    priv->has_texture_rg = vlc_gl_IsWebGL(&extension_vt) == false &&
+            (vlc_gl_GetVersionMajor(&extension_vt) >= 3
         || (interop->gl->api_type == VLC_OPENGL
             && vlc_gl_HasExtension(&extension_vt, "GL_ARB_texture_rg"))
         || (interop->gl->api_type == VLC_OPENGL_ES2
-            && vlc_gl_HasExtension(&extension_vt, "GL_EXT_texture_rg"));
+            && vlc_gl_HasExtension(&extension_vt, "GL_EXT_texture_rg")));
 
     video_color_space_t space;
     const vlc_fourcc_t *list;
-- 
2.43.0

